torchvision/prototype/ops/_utils.py

```diff
@@ -0,0 +1,119 @@
+import copy
+import operator
+import warnings
+from typing import Callable, Tuple, Union
+
+import torch
+from torch import fx
+from torchvision.models.feature_extraction import LeafModuleAwareTracer
+
+
+_MODULE_NAME = "_regularized_shotrcut"
+
+
+class RegularizedShortcut(torch.nn.Module):
+    def __init__(self, regularizer_layer: Callable[..., torch.nn.Module]):
+        super().__init__()
+        self._regularizer = regularizer_layer()
+
+    def forward(self, input, result):
+        return input + self._regularizer(result)
+
+
+def add_regularized_shortcut(
+    model: torch.nn.Module,
+    block_types: Union[type, Tuple[type, ...]],
+    regularizer_layer: Callable[..., torch.nn.Module],
+    inplace: bool = True,
+) -> torch.nn.Module:
+    if not inplace:
+        model = copy.deepcopy(model)
+
+    tracer = fx.Tracer()
+    modifications = {}
+    for name, m in model.named_modules():
+        if isinstance(m, block_types):
+            # Add the Layer directly on submodule prior tracing
+            # workaround due to https://github.com/pytorch/pytorch/issues/66197
+            m.add_module(_MODULE_NAME, RegularizedShortcut(regularizer_layer))
+
+            graph = tracer.trace(m)
+            patterns = {operator.add, torch.add, "add"}
+
+            input = None
+            for node in graph.nodes:
+                if node.op == "call_function":
+                    if node.target in patterns and len(node.args) == 2 and input in node.args:
+                        with graph.inserting_after(node):
+                            # Always put the shortcut value first
+                            args = node.args if node.args[0] == input else node.args[::-1]
+                            node.replace_all_uses_with(graph.call_module(_MODULE_NAME, args))
```

Comment:
Calling the previously created module by name. Hopefully this will be replaced with something like the following:

```python
fn_impl_traced = torch.fx.symbolic_trace(RegularizedShortcut(regularizer_layer))
args = node.args if node.args[0] == input else node.args[::-1]
fn_impl_output_node = fn_impl_traced(*map_arg(args, Proxy))
node.replace_all_uses_with(fn_impl_output_node.node)
```